libbeat/outputs/elasticsearch/client.go

```diff
@@ -329,48 +361,38 @@ func bulkEncodePublishRequest(
 }
 
 func createEventBulkMeta(
-	index outil.Selector,
+	indexSel outil.Selector,
 	pipelineSel *outil.Selector,
 	event *beat.Event,
-) interface{} {
+) (interface{}, error) {
 	pipeline, err := getPipeline(event, pipelineSel)
 	if err != nil {
-		logp.Err("Failed to select pipeline: %v", err)
+		err := fmt.Errorf("failed to select pipeline: %v", err)
+		return nil, err
 	}
 
-	if pipeline == "" {
-		type bulkMetaIndex struct {
-			Index   string `json:"_index" struct:"_index"`
-			DocType string `json:"_type" struct:"_type"`
-		}
-		type bulkMeta struct {
-			Index bulkMetaIndex `json:"index"`
-		}
-
-		return bulkMeta{
-			Index: bulkMetaIndex{
-				Index:   getIndex(event, index),
-				DocType: eventType,
-			},
-		}
+	index, err := getIndex(event, indexSel)
+	if err != nil {
+		err := fmt.Errorf("failed to select event index: %v", err)
+		return nil, err
 	}
 
-	type bulkMetaIndex struct {
-		Index    string `json:"_index" struct:"_index"`
-		DocType  string `json:"_type" struct:"_type"`
-		Pipeline string `json:"pipeline" struct:"pipeline"`
+	var id interface{}
+	if m := event.Meta; m != nil {
+		id = m["id"]
```

Comment:
Similar comment to other PR: Should we really allow here interface or make sure this is converted to a string? Do we know the key `"id"` exists in `m` ?

Reply:
can `_id` be a number instead of a string?

Reply:
I'm assuming that `_id` on the ES side is a keyword. So if `12` is put in it ends up also as `"12"`. Even if that is not the case, I think we should enforce string values on beats side.

Reply:
+1